app/genssz/types.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package main
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+)
+
+type Data struct {
+	Types []Type
+}
+
+type Type struct {
+	Name   string
+	Fields []Field
+}
+
+func (t Type) Abbr() string {
+	var lastUpper rune
+	for _, r := range t.Name {
+		if r >= 'A' && r <= 'Z' {
+			lastUpper = r
+		}
+	}
+
+	if lastUpper == 0 {
+		return strings.ToLower(t.Name[0:1])
+	}
+
+	return strings.ToLower(string(lastUpper))
+}
+
+type Field struct {
+	Index  int
+	Name   string
+	SSZTag string
+}
+
+func (f Field) IsComposite() bool {
+	return f.SSZTag == "Composite"
+}
+
+func (f Field) IsCompositeList() bool {
+	return strings.HasPrefix(f.SSZTag, "CompositeList")
+}
+
+func (f Field) IsByteList() bool {
+	return strings.HasPrefix(f.SSZTag, "ByteList")
+}
+
+func (f Field) IsBytesN() bool {
+	return strings.HasPrefix(f.SSZTag, "Bytes")
+}
+
+func (f Field) IsUint64() bool {
+	return f.SSZTag == "uint64"
+}
+
+func (f Field) MustSize() int {
+	size, err := f.Size()
+	if err != nil {
+		panic(err)
+	}
+
+	return size
+}
+
+func (f Field) Size() (int, error) {
+	var intStr string
+	if f.IsByteList() || f.IsCompositeList() {
+		openIdx := strings.Index(f.SSZTag, "[")
+		closeIdx := strings.Index(f.SSZTag, "]")
+		intStr = f.SSZTag[openIdx+1 : closeIdx]
```

Comment:
I suggest checking for presence of opening and closing brackets to avoid puzzling errors like this when the expression isn't well-formed: 

```
Fatal run error: write template: exec template: template: :48:76: executing "" at <.Size>: error calling Size: runtime error: slice bounds out of range [:-1]
```

```suggestion
		openIdx := strings.Index(f.SSZTag, "[")
		closeIdx := strings.Index(f.SSZTag, "]")

		if openIdx == -1 || closeIdx == -1 {
			return 0, errors.New("field has malformed size tag", z.Str("field_name", f.Name))
		}

		intStr = f.SSZTag[openIdx+1 : closeIdx]